src/subdomains/core/liquidity-management/adapters/actions/dfx-dex.adapter.ts

```diff
@@ -161,9 +161,13 @@ export class DfxDexAdapter extends LiquidityActionAdapter {
     const { system, assetId } = this.parseWithdrawParams(order.action.paramMap);
 
     const exchange = this.exchangeRegistry.get(system);
-    const asset = assetId ? await this.assetService.getAssetById(assetId) : order.pipeline.rule.targetAsset;
 
-    const deposits = await exchange.getDeposits(asset.dexName, order.created);
+    const sourceAsset = assetId ? await this.assetService.getAssetById(assetId) : undefined;
+    const sourceChain = sourceAsset && exchange.mapNetwork(sourceAsset.blockchain);
```

Comment:
Couldn't we use `sourceAsset ? exchange.mapNetwork(sourceAsset.blockchain) : undefined;` and then in line 170 we could replace `sourceChain || undefined` with `sourceChain`?